db.go

```diff
@@ -31,10 +32,38 @@ import (
 	"github.com/percona/qan-api2/migrations"
 )
 
+const (
+	databaseNotExistErrorCode = 81
+)
+
 // NewDB return updated db.
 func NewDB(dsn string, maxIdleConns, maxOpenConns int) *sqlx.DB {
 	db, err := sqlx.Connect("clickhouse", dsn)
 	if err != nil {
+		if exception, ok := err.(*clickhouse.Exception); ok && exception.Code == databaseNotExistErrorCode {
+			log.Println("Creating database")
+			clickhouseURL, err := url.Parse(dsn)
+			if err != nil {
+				log.Fatal(err)
+			}
+			q := clickhouseURL.Query()
+			databaseName := q.Get("database")
+			q.Set("database", "default")
+
+			clickhouseURL.RawQuery = q.Encode()
+
+			defaultDB, err := sqlx.Connect("clickhouse", clickhouseURL.String())
+			if err != nil {
+				log.Fatal("Connection: ", err)
+			}
+			result, err := defaultDB.Exec(fmt.Sprintf(`CREATE DATABASE %s ENGINE = Ordinary`, databaseName))
+			if err != nil {
+				log.Fatalf("Create database. Result: %v, Error: %v", result, err)
+			}
```

Comment:
this part can be extracted to separate function like 
```
func createDB(dsn string) error {
```
Return error in case of error instead of calling log.Fatal and close defaultDB connection in `defer`. 
in `newDB` after calling `createDB` in case of error is not nil we should call `log.Fatal(err)`.
```
err = createDB(dsn)
if err != nil {
    log.Fatalf("Database wasn't created: %s", err)
}
```
and after that we can reconnect to clickhouse
```
db, err = sqlx.Connect("clickhouse", dsn)
```